_rules/visible-label-in-accessible-name-2ee8b8.md

```diff
@@ -117,45 +123,242 @@ This `button` element has the text "search" rendered as an magnifying glass icon
 <button aria-label="Find">search</button>
 ```
 
+#### Passed Example 7 
+
+This button has [visible inner text][] that, according to the [label in name algorithm][], is contained within the [accessible name][].  This example shows why the [label in name algorithm][] uses the [visible inner text][] and not the [visible text content][]: the <p> tags insert whitespace into the result in the former but not the latter.
+
+```html
+<button aria-label="Hello world"><p>Hello</p><p>world</p></button>
+```
+
+#### Passed Example 8
+
+Similar to the previous example.
+
+```html
+<a href="#" aria-label="Some article by John Doe"><h6>Some article</h6><p>by John Doe</p></a>
+```
+
+#### Passed Example 9
+
+The [visible inner text][] is "Download specification".  The words "the" and "gizmo" aren't part of it.    
+
+```html
+<a aria-label="Download specification" href="#">Download <span style="visibility: hidden">the</span> <span style="display: none">gizmo</span> specification</a>
+```
+
+#### Passed Example 10
+
+This example shows that the [visible inner text][] isn't always the same as the [innerText][https://html.spec.whatwg.org/multipage/dom.html#the-innertext-idl-attribute].  The visible inner text is "Download specification".  The innerText is 'Download \ngizmo\nspecification'.  This rule uses the visible inner text - not innerText.
+
+```html
+<style>
+.visually-hidden {
+    /* Source: https://www.tpgi.com/the-anatomy-of-visually-hidden/ */
+    clip-path: inset(50%);
+    height: 1px;
+    overflow: hidden;
+    position: absolute;
+    white-space: nowrap;
+    width: 1px;
+}
+</style>
+<a aria-label="Download specification" href="#">Download <span class="visually-hidden">gizmo</span> specification</a>
+```
+
+#### Passed Example 11
+
+This example shows that the [label in name algorithm][] handles many kinds of whitespace.
+
+```html
+<a aria-label="compose email" href="#">compose &nbsp;&nbsp;<br> email</a>
+```
+
+#### Passed Example 12
+
+This example passes the rule because "YYYY-MM-DD" is in brackets.  Text in brackets is removed by the [label in name algorithm][], because its not normally spoken by speech-input users.  
+
+```html
+<button aria-label="Search by date">Search by date (YYYY-MM-DD)</button>
+```
+
+#### Passed Example 13
+
+The passes for two reasons: 1) because the ellipsis ("…") is [non-text content][], and 2) because the ellipsis is neither a letter nor a digit and so is filtered out by the [label in name algorithm][].
+
+```html
+<button aria-label="Next">Next…</button>
+```
+
+#### Passed Example 14
+
+This passes because the [label in name algorithm][] effectively ignores all punctuation and emoji, in both the visible inner text and the accessible name, as long as they don't break up words.
+
+```html
+<button aria-label="💡 Submit 💡">>>> ** Submit ** <<<</button>
+```
+
+#### Passed Example 15
+
+The "X" is non-text content.
+
+```html
+<button aria-label="Close">X</button>
+```
```

Comment:
```suggestion
```
(this is the same as Passed Example 5)

Reply:
Thanks, I just pushed a commit for this.